### PR TITLE
use fixed-width types when necessary

### LINKED ISSUE #29: 64-bit build creates bad ISOs
I'm not sure why yet but building in 64b mode creates a reproducible bad ISO with the "Halo 2 Multiplayer Map Pack" ~~and "Call of Duty: Big Red One"~~ game files (which I've ripped from my discs, verified to work on my real Xbox).

When running H2:MMP with such a faulty ISO, the game will start but it will fail to install update the game executable and report "out of space" errors when trying to install the maps. ~~In CoD:BRO, the game will report dirty disc errors before getting to the main menu.~~ (Fixed in #52)

In both cases, building in 32-bit mode yields a fully-functional ISO. Suggest updating CI to build in 32-bit mode until this can be fixed.

Thanks to @Ernegien for helping me debug this issue!

### PR DESCRIPTION
**Pull request by LukeUsher**
Use fixed-width types from stdint where it makes sense to do so.

Also fixes a couple of compiler warnings due to mismatching printf arguments on x64 builds.

Should fix #29. 

I don't have the games listed in that issue, but I have confirmed that resulting ISOs have matching sha1 sums when hardcoding ftimestamps (patch not included) to ensure reproducible images

Please verify the following before merging:

- [ ]  When running H2:MMP with such a faulty ISO, the game will start but it will fail to install update the game executable and report "out of space" errors when trying to install the maps.

- [ ] In CoD:BRO, the game will report dirty disc errors before getting to the main menu.

### CODE REVIEW

**Comment by dracc** [comment]
Code looks good, I have yet to test it.

File: extract-xiso.c [hunk 1]
```diff
@@ -1664,7 +1664,7 @@ int extract_file( int in_xiso, dir_node *in_file, modes in_mode , char* path) {
 				}
 					totalsize += size;
 					totalpercent = ( totalsize * 100.0 ) / in_file->file_size;
-					exiso_log( "%s%s%s (%lu bytes) [%lu%%]%s\r", in_mode == k_extract ? "extracting " : "", path, in_file->filename, in_file->file_size , totalpercent, "" );
+					exiso_log( "%s%s%s (%u bytes) [%u%%]%s\r", in_mode == k_extract ? "extracting " : "", path, in_file->filename, in_file->file_size , totalpercent, "" );
```
Comment (JayFoxRox) [inline extract-xiso.c:1667]: Consider `inttypes.h` (`PRIu32` and friends) so this can't causes issues in the future.

**Comment by abaire** [comment]
In case it's helpful, I just verified that this patch fixes the crash I ran into after fixing the root cause of mborgerson/xemu#509.

**Comment by mborgerson** [comment]
Thanks for the patch.

This does fix packing CoD:BRO, but does not resolve the issues with Halo 2 MMP.

I'll merge as it's improved, but I do not recommend using 64b builds of extract-xiso yet.

**Pull request merged**